### PR TITLE
Add Description of Validation Suites

### PR DESCRIPTION
**Pull request by bradtopol**

### CODE REVIEW

File: contributors/devel/sig-architecture/validation-suites.md [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+which instead refers to the SIG Architecture core set of interoperable features that
+all conformant Kubernetes clusters must support. Instead, Validation suites 
+validate correct behavior on an optional or extension capability.  Over time a
+a Validation suite or portion of it may be promoted to become a 
+[Conformance Test](../sig-architecture/conformance-tests.md) if deemed necessary. 
+
+## Validation Suite Test Requirements
+
+Validation Test Suites may be used as deemed necessary by the associated Kubernetes
+SIG that is defining the Validation Suite.
```
Comment (dims) [inline contributors/devel/sig-architecture/validation-suites.md]: defining and maintaining :)

Comment (dims) [reply contributors/devel/sig-architecture/validation-suites.md]: Unlike conformance tests, Validation Suites may change over time with addition/deletions/modifications as the SIG learns more about the problem space and adapt the implementations. Hence Validation Suites are not subject to the deprecation cycles (say for features) and left to the decision making process(es) of the SIG. The end users of the validation suites are strongly encouraged to participate in the upkeep and maintenance of the suite so they know what changes are coming up and can adapt their product/implementation to the updates of the test suite.

File: contributors/devel/sig-architecture/validation-suites.md [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+
+Validation Test Suites may be used as deemed necessary by the associated Kubernetes
+SIG that is defining the Validation Suite. 
+
+## Running Validation Suite Tests
+
+Validation Suite tests are designed to be run even when there is no cloud provider
+configured. Validation Suite tests must be able to be run against clusters that have
+not been created with `hack/e2e.go`, just provide a kubeconfig with the
+appropriate endpoint and credentials.
```
Comment (dims) [inline contributors/devel/sig-architecture/validation-suites.md]: testing-commons is tasked with adding a framework for writing test suites, so that work can be used for this

File: contributors/devel/sig-architecture/validation-suites.md [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+
+These commands are intended to be run within a kubernetes directory, either
+cloned from source, or extracted from release artifacts such as
+`kubernetes.tar.gz`. They assume you have a valid golang installation.
+
+```sh
+# ensure kubetest is installed
```
Comment (dims) [inline contributors/devel/sig-architecture/validation-suites.md]: 34-48, we don't have to dictate exactly how exactly each test suite has to run. We just have to say that there should be 
- easy to use, follow patterns similar to existing e2e tests
- enough documenation
- ability to select some tests and skip others

File: contributors/devel/sig-architecture/validation-suites.md [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+kubetest --provider=skeleton --test --test_args="--ginkgo.focus=\[Serial\].*\[*Validation\]"
+```
+
+Note that if you would like to run a specific Validation Suite instead of all the
+Validation Suites, replace `*Validation` in `--ginkgo.focus=\[*Validation\]` with
+a more specific Validation Suite name such as `StorageValidation`
+
+
+
+### Validation Suite Test Comment Metadata
```
Comment (dims) [inline contributors/devel/sig-architecture/validation-suites.md]: this section looks good

File: contributors/devel/sig-architecture/validation-suites.md [hunk 5]
```diff
@@ -0,0 +1,82 @@
+# Validation Suite Testing in Kubernetes
+
+A Kubernetes Validation suite is a subset of e2e tests that are used to
+validate correct functionality of an optional capability of Kubernetes. For example,
+Validation suites may be used to verify that optional storage or networking
```
Comment (timothysc) [inline contributors/devel/sig-architecture/validation-suites.md]: So this language overlaps with "Profiles".  We should probably refine the taxonomy.

**Comment by SeanPSnyder** [comment]
My agreement was waking up on a beach with an explanation not this shit
hole and a Gmail

File: contributors/devel/sig-architecture/validation-suites.md (hunk 5 above)
Comment (bradtopol) [reply contributors/devel/sig-architecture/validation-suites.md]: @timothysc Good point. I will revise and also add  some clarifying sentences that explain how validation suites are different from profiles and the flexibility they provide us to create suites of bucketized tests to provide validation independent of the process of defining/deciding that the validation suite should eventually become merged into the base profile, become their own profle etc.

Comment (quinton-hoole-2) [reply contributors/devel/sig-architecture/validation-suites.md]: +1

File: contributors/devel/sig-architecture/validation-suites.md [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+
+## Validation Suite Test Requirements
+
+Validation Test Suites may be used as deemed necessary by the associated Kubernetes
+SIG that is defining the Validation Suite.  Unlike Conformance Tests, Validation
+Suites may change over time with additions, deletions, and modifications whenever
+the SIG learns more about the problem space and adapts the implementations.
+Hence, Validation Suites are not subject to the deprecation cycles such as those 
+that apply to features and modification decisions are left to the decision making
+provess(es) of the SIG.  The end users of the Validation Suites are strongly
```
Comment (quinton-hoole-2) [inline contributors/devel/sig-architecture/validation-suites.md]: typo?  processes?

File: contributors/devel/sig-architecture/validation-suites.md [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+### Sample Validation Suite Test
+
+The following snippet of code shows a sample Storage Validation Suite test's 
+metadata.  Note how the test contains a bracketed label with name of the Validation
+Suite that this test belongs to. 
+
+```
+/*
+  Release : v1.14
+  Testname: CSI Attach, Non-attachable volume used in a Pod
```
Comment (quinton-hoole-2) [inline contributors/devel/sig-architecture/validation-suites.md]: nit: We might need to be more specific about the substructure of the name (if any). e.g. Is 'CSI Attach' a category of some sort?  Why is 'Non' capitalized?  Is this intended to be a proper sentence? e.g. "CSI-Attach (used as a verb) a non-attachable volume (object) used in a Pod (qualifying clause).

In summary, perhaps this example nname is a bad one, unless you explain why it's pretty weird-looking.

Comment (quinton-hoole-2) [reply contributors/devel/sig-architecture/validation-suites.md]: nit: PS: Why not just 'Name' instead of 'Testname'?

File: contributors/devel/sig-architecture/validation-suites.md [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+  Description: When CSI driver attach is called on non attachable volumes it should not create a volume attachment.
+*/
+It("should not require VolumeAttach for drivers without attachment [StorageValidation]", func() {
```
Comment (quinton-hoole-2) [inline contributors/devel/sig-architecture/validation-suites.md]: nit: This does not seem to correspond with the description or name given above, so again, the example is quite confusing.

File: contributors/devel/sig-testing/e2e-tests.md [hunk 9]
```diff
@@ -25,6 +25,7 @@
   - [Kinds of tests](#kinds-of-tests)
     - [Viper configuration and hierarchichal test parameters.](#viper-configuration-and-hierarchichal-test-parameters)
     - [Conformance tests](#conformance-tests)
+	- [Validation suites](#validation-suites)
```
Comment (quinton-hoole-2) [inline contributors/devel/sig-testing/e2e-tests.md]: The structure here implies that you consider a Validation Suite to be a kind of Conformance Test?  If so, I think that should be made more clear in the body of the document, because currently it's not.

Comment (quinton-hoole-2) [reply contributors/devel/sig-testing/e2e-tests.md]: Either that, or change the structure to be more consistent with the statement in the doc that "a Validation Suite is a subset of e2e tests".

File: contributors/devel/sig-architecture/validation-suites.md [hunk 10]
```diff
@@ -0,0 +1,82 @@
+# Validation Suite Testing in Kubernetes
```
Comment (quinton-hoole-2) [inline contributors/devel/sig-architecture/validation-suites.md]: Overall comment, and extending what @timothysc mentioned below:
I find the general taxonomy of e2e tests to be confusing.  Even if we ignore "Profiles" for the moment, the relationship between:

1. e2e tests (generally)
2. Validation test suites
3. Conformance tests
4. All the other categories and sub-categories of e2e tests that we have

is not at all clear to me based on the descriptions given in this doc.  This is further compounded by the somewhat ambiguous naming -- the dictionary distinction between 'Validation' ("the action of making or declaring something legally or officially acceptable") and 'Conformance' ("compliance with standards, rules, or laws") is very blurry indeed.

I added more comments around this below.

File: contributors/devel/sig-architecture/validation-suites.md [hunk 11]
```diff
@@ -0,0 +1,82 @@
...
+all conformant Kubernetes clusters must support. Instead, Validation suites 
+validate correct behavior on an optional or extension capability.  Over time a
+a Validation suite or portion of it may be promoted to become a 
+[Conformance Test](../sig-architecture/conformance-tests.md) if deemed necessary. 
+
+## Validation Suite Test Requirements
+
+Validation Test Suites may be used as deemed necessary by the associated Kubernetes
+SIG that is defining the Validation Suite.  Unlike Conformance Tests, Validation
+Suites may change over time with additions, deletions, and modifications whenever
```
Comment (quinton-hoole-2) [inline contributors/devel/sig-architecture/validation-suites.md]: I find this description rather confusing, possibly because I simply disagree with it.

Surely we should have optional functionality which is none-the-less GA, and subject to all the same backwards compatibility guarantees as non-optional GA features?  And this admittedly probably bleeds into the profiles discussion.

If I as a user decide to use an optional capability (optional in the sense that it's allowable to have a cluster called Kubernetes even if it does not provide the capability, like remote attached storage), that is GA (in the sense that the API is fixed) I don't want the definition of how remote attached storage works to change from one release to the next?  Probably better discussed further on a call.

Comment (dims) [reply contributors/devel/sig-architecture/validation-suites.md]: @quinton-hoole-2 this takes into account how nodeconformance, csi test suite etc grew over time. note that even CRI is v1alpha2 (if i remember right). 

If you need guarantees, then that would be a conformance suite ... not a validation suite.

**Comment by quinton-hoole-2** [comment]
/unassign quinton-hoole-2

Unassigned myself since I finished one round of review.  Please re-assign to me if you require another review.

File: contributors/devel/sig-architecture/validation-suites.md (hunk 11 above)
Comment (quinton-hoole-2) [reply contributors/devel/sig-architecture/validation-suites.md]: @dims That may be true, but that's not what this document says.  

"A Kubernetes Validation suite is a subset of e2e tests that are used to
validate correct functionality of an optional capability of Kubernetes."

Comment (dims) [reply contributors/devel/sig-architecture/validation-suites.md]: @quinton-hoole-2 may be the missing piece is "for use with a specific release"? (or subject to change across release boundaries)

Comment (quinton-hoole-2) [reply contributors/devel/sig-architecture/validation-suites.md]: Perhaps we have 3 separate axes that we're confusing in this doc:

1. optional vs non-optional
2. alpha vs beta vs GA
3. Kubernetes release number

a. The degree to which tests can change over time has to do with the axes 2 and 3.  If a test passes for a GA feature in release number x, then that same test should continue to pass, unchanged, essentially for all releases subsequent to x (due to backwards compatibility guarantees).
b. If release number x+n adds additional optional fields to an API, then a (logically) new test (to validate that those fields are correctly handled if present) for release x+n is required, and depending on whether the fields are alpha, beta or GA, the test may be required to pass for all releases subseqent to x+n.
c. The degree to which a test is required to pass in order for a cluster to be conformant has to do with axis 1.

Thoughts?

File: contributors/devel/sig-architecture/validation-suites.md (hunk 6 above)
Comment (bradtopol) [reply contributors/devel/sig-architecture/validation-suites.md]: yes, a typo. Will fix.

File: contributors/devel/sig-architecture/validation-suites.md (hunk 7 above)
Comment (bradtopol) [reply contributors/devel/sig-architecture/validation-suites.md]: @quinton-hoole-2 the above was a verbatim sample taken from one test in a new validation suite that @brahmaroutu added. I believe he is keeping the syntax consistent with conformance tests to make promotion simpler.

Comment (brahmaroutu) [reply contributors/devel/sig-architecture/validation-suites.md]: @quinton-hoole-2 We use this same mechanism on Conformance Suite, a TestName is currently parsed by a tool that can generate conformance documentation, we can use the same tool here.  
You are right that the Test name has a category that defines the what is being tests ex: PodSpec or CSI Attach, etc. The description part is a free form text and need no be capitalized.
We can change the Example if it is confusing WDYT?

File: contributors/devel/sig-architecture/validation-suites.md (hunk 10 above)
Comment (bradtopol) [reply contributors/devel/sig-architecture/validation-suites.md]: I'm adding more clarity to distinguish between validation and conformance, incorporating many of the explanations provided by @dims

File: contributors/devel/sig-architecture/validation-suites.md (hunk 11 above)
Comment (bradtopol) [reply contributors/devel/sig-architecture/validation-suites.md]: Good point made above. I will address in the next version.

File: contributors/devel/sig-testing/e2e-tests.md (hunk 9 above)
Comment (bradtopol) [reply contributors/devel/sig-testing/e2e-tests.md]: Structure is wrong.  They are peers. Validation is not a kind of conformance test.  I will fix

Comment (bradtopol) [reply contributors/devel/sig-testing/e2e-tests.md]: I had a hidden tab in there :-). It's been removed

File: contributors/devel/sig-architecture/validation-suites.md (hunk 7 above)
Comment (quinton-hoole-2) [reply contributors/devel/sig-architecture/validation-suites.md]: @brahmaroutu I think just put in a link to the documentation explaining the weird formatting of the test name.

File: contributors/devel/sig-architecture/validation-suites.md (hunk 10 above)
Comment (quinton-hoole-2) [reply contributors/devel/sig-architecture/validation-suites.md]: FYI, I'm not sure I agree with @Dims on that 😁

Comment (dims) [reply contributors/devel/sig-architecture/validation-suites.md]: LOL sometimes i don't agree with myself :)

**Comment by dims** [comment]
Let's try to do one more round of review on this (this week) please? @quinton-hoole

File: contributors/devel/sig-architecture/validation-suites.md [hunk 12]
```diff
@@ -0,0 +1,95 @@
...
+validate functionality of an optional capability of Kubernetes for use
+with a specific release. Validation suites can grow or shrink over time, and can also
+change as a functionality progresses from alpha, to beta, and to GA. The flexibility
+of Validation suites to change based on experimentation and feedback is a key 
+characteristic that differentiates Validation suites from 
+[Conformance Tests](../sig-architecture/conformance-tests.md) and 
+[Conformance profiles](https://github.com/cncf/k8s-conformance/blob/master/terms-conditions/Certified_Kubernetes_Terms.md). In addition, Validation suites differ from  
+Conformance profiles in that they do not require conformance certification approval
+in order to be created and utilized.  The creation of a Conformance Profile requires
+consensus and discussion across the community because there is a desire to limit
```
Comment (dims) [inline contributors/devel/sig-architecture/validation-suites.md]: "The creation of a Conformance Profile requires
consensus and discussion across the community because there is a desire to limit
the number of conformance profiles created; a large number of Conformance Profiles
would inhibit the portability of Kubernetes workloads."

This is unrelated to subject at hand @bradtopol can we please remove this?

**Comment by dims** [comment]
One comment inside, let's remove the unnecessary verbiage around conformance profiles and we can request @bgrant0607 and @quinton-hoole to look after that

File: contributors/devel/sig-architecture/validation-suites.md (hunk 12 above)
Comment (bradtopol) [reply contributors/devel/sig-architecture/validation-suites.md]: Hi @dims.  Sure. Will do that right now

File: contributors/devel/sig-architecture/validation-suites.md (hunk 10 above)
Comment (bgrant0607) [reply contributors/devel/sig-architecture/validation-suites.md]: Catching up. It looks like this still wasn't resolved?

We need to tests interfaces of extension points and of individual pluggable components. Many of those won't be end-to-end tests. That's what I thought we would call validation suites.

Conformance tests are a subset of end-to-end tests.

Comment (brahmaroutu) [reply contributors/devel/sig-architecture/validation-suites.md]: @bgrant0607  Sorry I am trying to understand this. If you do not want to call them as 'subset of end-to-end' tests, my assumption was that we could trigger a validation suite by just using a proper regex in the ginkgo.focus. We can refine the statement saying 'Validation Suite tests are similar e2e tests and may be part of or extension to existing e2e tests'?

File: contributors/devel/sig-architecture/validation-suites.md (hunk 11 above)
Comment (johnbelamaric) [reply contributors/devel/sig-architecture/validation-suites.md]: @bradtopol I don't think this was ever addressed

File: contributors/devel/sig-architecture/validation-suites.md [hunk 13]
```diff
@@ -0,0 +1,92 @@
+# Validation Suite Testing in Kubernetes
+
+A Kubernetes Validation suite is a subset of e2e tests that are used to
+validate functionality of an optional capability of Kubernetes for use
+with a specific release. Validation suites can grow or shrink over time, and can also
```
Comment (johnbelamaric) [inline contributors/devel/sig-architecture/validation-suites.md]: I think it should be clear that, like conformance tests, these are tests that are intended to be used outside of the standard CI, and run against an arbitrary cluster. They are used to validate that a given provider/distribution implements a feature in the standard way. The goal in my mind is that a validation suite should allow us to enforce that a given feature works in the same manner across distributions. If that's the goal (and it is left to, say, profiles), then we should also explicitly state that. But I am not sure what the point is if that's not the goal.

Comment (bradtopol) [reply contributors/devel/sig-architecture/validation-suites.md]: Hi @johnbelamaric I think there is a lot of history on this topic you should catch up on. We covered this at the f2f meeting at KubeCon Seattle https://youtu.be/dueJApGGAo0
But essentially validation suites were not something that was mandatory for each distribution to run. We needed a way for both vendors and  the e2e test team to create groups of test based on functionality that was optional.  The conformance tests and possibly the creation of conformance  profiles is something that takes a large amount of discussion and consensus.  Validation suites are created and controlled by SIGs and can be experimental and can change.  Conformance tests are more official and tied to the conformance certification program.  @dims had good examples for why we should do validation suites first.

Comment (johnbelamaric) [reply contributors/devel/sig-architecture/validation-suites.md]: Ok. Well, if you can address the other comments maybe we can get this merged soon?

**Comment by spiffxp** [comment]
/unassign
I'm more interested in core conformance

File: contributors/devel/sig-architecture/validation-suites.md (hunk 13 above)
Comment (bradtopol) [reply contributors/devel/sig-architecture/validation-suites.md]: @johnbelamaric This was handled by adding clarity about what validation suites actually were used for and how they were different from conformance profiles. The portions where I discuss how validation suites can grow and shrink over time and their flexibility of a validation suite to change based on experimentation and feedback were the key characteristics that differentiated the validation suites from conformance profiles.  Between this and removing the section that @dims asked to be removed I believe I addressed  the concerns that @quinton-hoole had regarding the taxonomy.  So unless folks still don't understand how validation suites are different than conformance profiles due to the characteristics I describe, I believe this is ready to merge.

File: contributors/devel/sig-architecture/validation-suites.md [hunk 14]
```diff
@@ -0,0 +1,92 @@
...
+[Conformance profiles](https://github.com/cncf/k8s-conformance/blob/master/terms-conditions/Certified_Kubernetes_Terms.md). In addition, Validation suites differ from  
+Conformance profiles in that they do not require conformance certification approval
+in order to be created and utilized. As a result, Validation 
+suites can be created to validate the functionality of an optional 
+capability and the more lengthy community discussion of whether the validation suite
+should be promoted to become a Conformance profile or merged into the base
+Conformance Tests can be deferred.  Validation suites allow the Kuberentes e2e
+testing team to rapidly create test suites for a variety of use cases.  They may
+also be used to identify a group of tests that validate functionality on an
+optional platform such as Kuberenetes on Windows.  Kubernetes SIG groups are
```
Comment (brahmaroutu) [inline contributors/devel/sig-architecture/validation-suites.md]: @bradtopol I think validation suites should be targeted to functionality but not at platforms. Can you please remove this sentence with Windows as an example.

File: contributors/devel/sig-architecture/validation-suites.md (hunk 13 above)
Comment (brahmaroutu) [reply contributors/devel/sig-architecture/validation-suites.md]: @bradtopol Please change the line#1 to not say 'subset of e2e tests'

Comment (bradtopol) [reply contributors/devel/sig-architecture/validation-suites.md]: Wiil do!

File: contributors/devel/sig-architecture/validation-suites.md (hunk 14 above)
Comment (bradtopol) [reply contributors/devel/sig-architecture/validation-suites.md]: Will do!

**Comment by timothysc** [comment]
So I re-read this doc and I'm still confused by how this differs from "profiles".

**Comment by bradtopol** [comment]
> So I re-read this doc and I'm still confused by how this differs from "profiles".

@timothysc @johnbelamaric So good question. The difference is staging and speed and the amount of time needed for proper vetting.  Let me provide a real world example to make this clear:

So @brahmaroutu wants to investigate what is the proper set of validation tests for something like storage.  He wants to get started today.  He wants to work with SIG Storage and iterate with them on  what is the proper suite of validation tests.  This could take some time as he and the team iterate and tests may be added and removed until they get to what he and SIG Storage agree is the proper set of validation tests.

Now at this point in time @brahmaroutu and SIG storage may believe the test suite should become part of conformance.  If we are taking this process seriously we would then kick off a conformance discussion that looks at whether this new suite of tests should be added to core conformance or if there is sufficient justification to agree that the storage tests should become a new conformance profile.  We should never want the default and easy answer to be "sure just create a new conformance profile"   If you have to worry about making sure a cloud infrastructure and its workloads  had to be interoperable/portable across multiple Kubernetes platforms/distributions, the concept of those platforms/distributions supporting multiple combinations of those conformance profiles would scare you to death.   In a previous life,  I was the interoperability lead for a different  cloud infrastructure and responsible for ensuring workloads were portable across 15+ platforms/distributions and I also had to demonstrate this in a live conference keynote (see https://www.youtube.com/watch?v=CqNFcuJfZ_Y)  I simply cannot imagine pulling this off and being successful if all 15 platforms/distributions supported different combinations of several conformance profiles. The old joke that "the great thing about standards is there is so many to choose from"  really begins to apply here. 

Instead, the only way to be successful, would be to 1) always try first to add a profile to core conformance and if that is impossible  then 2) at least group multiple conformance profiles together and agree which of those will be supported by equivalence classes of platforms/distros so that to the end user it appears as if the majority of the platforms/distributions really are providing a single conformance suite.

Now the process I described above for properly vetting whether a suite of tests can be moved into core conformance or if a new conformance profile is truly required takes proper vetting and consensus building in the community.  All of that is important but if you are someone like @brahmaroutu that just wants to investigate and iterate on what is the proper set of validation test for something like storage, you do not want to be slowed down by all the vetting/discussion/consensus building that by necessity should occur when making an important decision regarding what becomes a conformance profile.   So as an effort to let test creators be test creators and to not become slowed down or disenchanted by the conformance profile decision process, we defined the concept of validation suites that allows those important sets of validation tests to be created and all the more time consuming discussions that test developers hate can be done without slowing them down. Hope this helps. Also we discussed the issue at the f2f in KubeCon Seattle and that discussion is documented here: https://youtu.be/dueJApGGAo0

Thanks for taking the time to read all this.

**Comment by quinton-hoole** [comment]
Thanks @bradtopol and others for all the work that has gone into this. I've re-read it after a long hiatus (apologies for that).  Thanks for addressing all of my original comments.  I think I have a much better understanding of your goals here now.  

I have one remaining, overarching question (and be warned, it's a rather large one).
As you describe it, the "Validation suite" distinction doesn't seem to actually mean anything significant.  If I may paraphrase/summarize, according to this proposal, an e2e test that is part of a given "Validation Suite":

1. Is not required for conformance.
2. Has not been formally agreed upon (e.g. by sig-arch) as the standard, long-term API or behavior of a (possibly optional) feature.
3. Forms part of a loosely defined grouping of e2e tests that may change over time.

If the above is true, why not just call the test an "e2e test", and the grouping a "group of e2e tests, defined by an arbitrary label"?

As an aside, I think that we have a significant number of other logical groupings of tests.  One of these, as @bgrant0607 points out, is tests to verify conformance of extension points (and extensions).  Another is to verify conformance to optional API functionality (i.e. if the optional feature exists, it's behavior conforms).  And there are plenty of others.

So here's a concrete proposal for a way forward:

1. We define a generalized categorization and labelling/taxonomy scheme for tests, together with a gatekeeper function to keep it somewhat sane/tidy (e.g. each category is owed by a sig, which approves changes to it's categories.  Each category would require a name and clear description).
2. We kickstart by pre-populating the sig-owned categories we think we need.  Maybe "Storage/Optional/Validation" (what @brahmaroutu wants) is one, and "Extensions/Storage" (part of wat @bgrant0607 wants),  "OS/Windows/Mandatory" (what @PatrickLang wants) and "OS/Windows/Optional" are others.  I just totally made those up, but hopefully the concept makes sense?

Thoughts?

**Comment by quinton-hoole** [comment]
We discussed this proposal regarding validation suites and decided not to proceed with it.  Closing.

/close
